sdlib/d/gc/tcache.d

```diff
@@ -421,3 +450,94 @@ unittest getCapacity {
 	assert(p5 !is p4);
 	assert(threadCache.getCapacity(p5[0 .. 16000]) == 20480);
 }
+
+unittest extend {
+	auto nonAppendable = threadCache.alloc(100, false);
+
+	// Attempt to extend a non-appendable:
+	assert(!threadCache.extend(nonAppendable[0 .. 100], 1));
+
+	// Make an appendable alloc:
+	auto p0 = threadCache.allocAppendable(100, false);
+	assert(threadCache.getCapacity(p0[0 .. 100]) == 16384);
+
+	// Attempt to extend slices with capacity 0:
+	assert(threadCache.extend(p0[0 .. 0], 0));
+	assert(!threadCache.extend(p0[0 .. 0], 50));
+	assert(!threadCache.extend(p0[0 .. 99], 50));
+	assert(!threadCache.extend(p0[1 .. 99], 50));
+	assert(!threadCache.extend(p0[0 .. 50], 50));
+
+	// Attempt extend with insufficient space:
+	assert(!threadCache.extend(p0[0 .. 100], 16285));
+	assert(!threadCache.extend(p0[50 .. 100], 16285));
+
+	// Extend by size zero is permitted but has no effect:
+	assert(threadCache.extend(p0[100 .. 100], 0));
+	assert(threadCache.extend(p0[0 .. 100], 0));
+	assert(threadCache.getCapacity(p0[0 .. 100]) == 16384);
+	assert(threadCache.extend(p0[50 .. 100], 0));
+	assert(threadCache.getCapacity(p0[50 .. 100]) == 16334);
+
+	// Extend by zero is permitted even when no capacity:
+	assert(threadCache.extend(nonAppendable[0 .. 100], 0));
+
+	void* nullPtr = null;
+	assert(threadCache.extend(nullPtr[0 .. 100], 0));
+
+	void* stackPtr = &nullPtr;
+	assert(threadCache.extend(stackPtr[0 .. 100], 0));
+
+	void* tlPtr = &threadCache;
+	assert(threadCache.extend(tlPtr[0 .. 100], 0));
```

Comment:
Don't put these int he middle of tests related to p0, and extending by other values should fail, but this isn't tested.